### DISCUSSION TITLE
Session in DB table is not used when authenticating

### DISCUSSION THREAD

**Discussion by workatease** [directus/directus#4337] [Archive]
Hi Team,

I am working on the authentication and to force logoff user session.

I see that the users session is not considered when we authenticate the users in below code as the direct_seesions tables is not considered.

https://github.com/directus/directus/blob/main/api/src/middleware/authenticate.ts#L39-L47

so the above block just make sure the jwt token is valid and has the users register in the DB but not the session.
is this by design or is my understanding is wrong?

we logout the users by deleting the user from the session table https://github.com/directus/directus/blob/main/api/src/services/authentication.ts#L191-L193
but if if we are not checking it it the above block are we missing the logic there?

**Comment by rijkvanzanten** [comment]
The idea of the JWTs for access tokens it that we only check the authentication at the start (when creating the token), and rely on the data stored in the token itself for requests made with the access token. This also means that we don't check authentication status on use of the token itself. As a matter of fact, the problem in that first link is that the API currently checks the user with the DB on every request with a JWT. It shouldn't do this.

The idea is that you rely on the contents of the JWT for every request, and only do the full authentication flow on login and refresh. This combined with the short life-span of the access token means that you can have a secure session _and_ the performance benefit of not going to the database for those checks on every request.

(So TL;DR, this is by design, and the API currently does _too much_ which we can optimize :) )

**Reply by workatease** [reply]
so logout options is not a actual logout if the access token is still valid till the expire time.
what do you think is the production ready expire time 
I have a use case of more than 2000 users and peak time count goes to near 400 users.

**Reply by rijkvanzanten** [reply]
It depends on what you consider "logging out". In this case, you start a session, in which you're allowed to retrieve access tokens that can be used to retrieve and modify data. On "logout", this session is destroyed, no longer allowing you to retrieve new access tokens. auth0 wrote up a good brief overview of this setup: https://auth0.com/blog/refresh-tokens-what-are-they-and-when-to-use-them/

> what do you think is the production ready expire time

Depends on how you're handling the tokens, what permission-level your authenticated user is, but also how powerful your server environment is. If you make the access token shorter, you'll have to hit the more costly /refresh endpoint more often, so with _a lot_ of users, that might cause some unwanted CPU usage. Generally speaking, 15 minutes is a pretty OK default value, and you can go up/down from there using [`ACCESS_TOKEN_TTL`](https://docs.directus.io/reference/environment-variables/#security)

**Reply by workatease** [reply]
Thank you @rijkvanzanten. your explanation help me understand it
